### PR TITLE
GO-2827 Unbind user files (reclaim / reconcile space usage)

### PR DESCRIPTION
**Pull request by deff7**

### CODE REVIEW

File: core/filestorage/filesync/remove.go [hunk 1]
```diff
@@ -96,3 +117,36 @@ func (s *fileSync) removeFromDeletionQueues(item *QueueItem) error {
 	}
 	return nil
 }
+
+func (s *fileSync) CancelDeletion(objectId string, fileId domain.FullFileId) error {
```
Comment (deff7) [inline core/filestorage/filesync/remove.go:121]: Test

File: util/keyvaluestore/store.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"fmt"
+
+	"github.com/dgraph-io/badger/v4"
+
+	"github.com/anyproto/anytype-heart/util/badgerhelper"
+)
+
+var ErrNotFound = fmt.Errorf("not found")
+
+type Store[T any] interface {
```
Comment (deff7) [inline util/keyvaluestore/store.go:13]: Comment

File: core/filestorage/filesync/upload.go [hunk 3]
```diff
@@ -142,75 +148,91 @@ func (s *fileSync) retryingHandler(ctx context.Context, it *QueueItem) (persiste
...
-	if s.onUploaded != nil {
-		err := s.onUploaded(fileObjectId)
-		if err != nil && !errors.Is(err, spacestorage.ErrTreeStorageAlreadyDeleted) {
-			log.Warn("on upload callback failed",
-				zap.String("fileObjectId", fileObjectId),
-				zap.String("spaceID", spaceId),
-				zap.Error(err))
-			return err
+func (s *fileSync) runOnUploadedHook(fileObjectId string, fileId domain.FullFileId) error {
+	var merr multierror.Error
```
Comment (cheggaaa) [inline core/filestorage/filesync/upload.go:191]: Let's use std errors.Join instead of multierr

Comment (deff7) [reply core/filestorage/filesync/upload.go:191]: done

**Review by fat-fellow — APPROVED** [review]

**Review by cheggaaa — APPROVED** [review]

**Pull request merged**